File: vocaflip-backend/src/main/java/com/vocaflipbackend/dto/response/StudyCardResponse.java
package com.vocaflipbackend.dto.response;

import com.vocaflipbackend.enums.LearningStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


// DTO thông tin thẻ trong phiên học. (dữ liệu thẻ gốc + trạng thái tiến độ hiện tại)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudyCardResponse {

    // Thông tin thẻ
    private String cardId;
    private String front;
    private String back;
    private String phonetic;
    private String exampleSentence;
    private String audioUrl;
    private String imageUrl;
    private Integer orderIndex;

    // Trạng thái tiến độ hiện tại
    private LearningStatus learningStatus;
    private Integer currentInterval;
    private Integer reviewCount;
}

File: voca_flip_mobile/lib/data/models/responses/study_card_response.dart
class StudyCardResponse {
  final String cardId;
  final String front;
  final String back;
  final String? phonetic;
  final String? exampleSentence;
  final String? imageUrl;
  final String? audioUrl;
  final int? orderIndex;

  const StudyCardResponse({
    required this.cardId,
    required this.front,
    required this.back,
    this.phonetic,
    this.exampleSentence,
    this.imageUrl,
    this.audioUrl,
    this.orderIndex,
  });

  /// JSON -> StudyCardResponse
  factory StudyCardResponse.fromJson(Map<String, dynamic> json) {
    return StudyCardResponse(
      cardId: json['cardId'] as String,
      front: json['front'] as String,
      back: json['back'] as String,
      phonetic: json['phonetic'] as String?,
      exampleSentence: json['exampleSentence'] as String?,
      imageUrl: json['imageUrl'] as String?,
      audioUrl: json['audioUrl'] as String?,
      orderIndex: json['orderIndex'] as int?,
    );
  }

  /// StudyCardResponse -> JSON
  Map<String, dynamic> toJson() {
    return {
      'cardId': cardId,
      'front': front,
      'back': back,
      'phonetic': phonetic,
      'exampleSentence': exampleSentence,
      'imageUrl': imageUrl,
      'audioUrl': audioUrl,
      'orderIndex': orderIndex,
    };
  }
}
